File: Graduater/Core/Entities/Poll.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities;

public class Poll : DatabaseEntity
{
    public string Title { get; set; } = null!;

    public string Description { get; set; } = null!;

    public int CreatorUserId { get; set; }

    public DateTime Due { get; set; }

    public DateTime DateCreated { get; set; }

    public bool IsAnonymous { get; set; }

    [ForeignKey(nameof(CreatorUserId))]
    public virtual User CreatorUser { get; set; } = null!;

    public virtual ICollection<PollOption> PollOptions { get; set; } = new List<PollOption>();
}

File: models/Poll.ts
import DataBaseEntity from "./DataBaseEntity";
import PollOption from "./PollOption";
import User from "./User";

export default interface Poll extends DataBaseEntity{
    CreatorUser?: User;
    creatorUserId: number;
    dateCreated: Date;
    description: string;
    due: Date;
    isAnonymous: boolean;
    pollOptions?: PollOption[];
    title: string;
}